### PR TITLE
ci: update spotbugs version for gradle 7+, java-17, java-20

### PR DESCRIPTION
**Pull request by soloturn**
spotbugs needs to be able to read new class format.

### CODE REVIEW

File: build-logic/src/main/kotlin/terasology-metrics.gradle.kts [hunk 1]
```diff
@@ -93,9 +93,6 @@ configure<PmdExtension> {
 }
 
 configure<SpotBugsExtension> {
-    // The version of the spotbugs tool https://github.com/spotbugs/spotbugs
-    // not necessarily the same as the version of spotbugs-gradle-plugin
-    toolVersion.set("4.7.0")
```
Comment (jdrueckert) [inline build-logic/src/main/kotlin/terasology-metrics.gradle.kts:98]: Why do remove this instead of updating it?

Comment (soloturn) [reply build-logic/src/main/kotlin/terasology-metrics.gradle.kts:98]: setting it gave an unexpected error, and if one NEEDS to anyway update it with a different gradle version, better use the default version by not setting it. if somebody feels to set the version again for a yet unknown reason, it is not too much work to add a line again.

Comment (soloturn) [reply build-logic/src/main/kotlin/terasology-metrics.gradle.kts:98]: test with
```
gradle clean spotbugsMain
....

❯ ls -l engine/build/reports/spotbugs/main.xml 
-rw-r--r-- 1  4585709 21. Aug 08:28 engine/build/reports/spotbugs/main.xml

```

**Comment by soloturn** [comment]
updated version to 5.1.3 released meanwhile, thanks to mention the version, @PurityLake , in #5130 .

File: build-logic/src/main/kotlin/terasology-metrics.gradle.kts (hunk 1 above)
Comment (jdrueckert) [reply build-logic/src/main/kotlin/terasology-metrics.gradle.kts:98]: ah okay, wasn't aware it was overriding the default version.
generally I'd rather know which version we're using, but considering our track record of updating dependencies, I guess it's fine to just go with whatever default the latest consumed gradle plugin version implies :+1:

**Review by jdrueckert — APPROVED** [review]

**Comment by jdrueckert** [comment]
fyi we're facing a bit of a problem with our jenkins today. once that's resolved, this can be merged.

**Comment by jdrueckert** [comment]
http://newerjenkins.terasology.io/job/Terasology/job/engine/job/PR-5126/7/ succeeded save for a sporadically failing integration test, so I merged this despite the checks not being green - we're currently facing some issues with Jenkins cloning the engine repo.

**Pull request merged**